### DISCUSSION TITLE
Scraping an open chrome tab

### DISCUSSION THREAD

**Discussion by antoniogiroz** [johnlindquist/kit#946] [Q&A] [answered]
Hello!

I want to scrape the content of a webpage which it's currently open in the browser, but I can't use the playwright or the `scrapeSelector` function because I need to stay logged in to access to the content.

Any tip to get the content of a chrome tab?

Thanks a lot!

**Comment by johnlindquist** [comment, marked as answer]
Yeah, for sure! Check out this answer (if you're on osx). I don't know of a way currently on Windows without playwright and logging in each time.

https://github.com/johnlindquist/kit/discussions/695

**Reply by antoniogiroz** [reply]
I've tested the proposed solution and the `value` is empty, but I execute the JS code in the tab console and it works. Any idea?

The only thing I've changed is the selector.

**Reply by antoniogiroz** [reply]
This is the code I'm using: 

```ts
import '@johnlindquist/kit';

let js = `
document.querySelector('h1[data-testid="issue.views.issue-base.foundation.summary.heading"]').innerText
`;

let value = await applescript(`
tell application "Google Chrome" to tell window 1
	set str to execute active tab javascript "${js}"
	return str
end tell
`);

await div(md(`## ${value}`));

// To copy to clipboard, use:
copy(value);
```

**Reply by johnlindquist** [reply]
Did you enable the dev JavaScript from Apple Events mentioned in that post?

**Reply by antoniogiroz** [reply]
Yes. Maybe do I need to set some permission to something?

**Reply by johnlindquist** [reply]
![CleanShot 2022-11-14 at 14 06 19](https://user-images.githubusercontent.com/36073/201765533-d3fb5019-3dfe-4d4d-beb1-44d1920911a3.png)

Yeah, I think it's that "Google Chrome" one. A lot of AppleScript scripts are for "System Events" too.

**Reply by antoniogiroz** [reply]
All is checked!

But I've tested with another css selector (with classes) and it works. Do you know some limitation with querying by attribute? 🤔

**Reply by johnlindquist** [reply]
Oh, I didn't look at your example, sorry. Try double-escaping your double quotes since you're injecting a string into a string:

`document.querySelector('div[role=\\"navigation\\"]').innerText`

**Reply by antoniogiroz** [reply]
Yeah! 😅 It works!

Thanks!
